src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,389 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using static Microsoft.DotNet.Interactive.CSharpProject.BuildCacheFileUtilities;
+
+namespace Microsoft.DotNet.Interactive.CSharpProject;
+
+public static class RoslynWorkspaceUtilities
+{
+    public class ProjectItem
+    {
+        public string ItemSpec { get; }
+        public IReadOnlyDictionary<string, string> Metadata { get; }
+
+        internal ProjectItem()
+        {
+//            ItemSpec = taskItem.ItemSpec;
+//            Metadata = taskItem.MetadataNames.Cast<string>().ToDictionary(x => x, x => taskItem.GetMetadata(x));
+        }
+    }
+
+    public class BuildDataResults
+    {
+        public BuildProjectData Data { get; set; }
+
+        public Microsoft.CodeAnalysis.Workspace Workspace { get; set; }
+
+        public bool Succeeded { get; set; }
+
+        public string cacheFile { get; set; }
```

Comment:
```suggestion
        public string CacheFile { get; set; }
```

Reply:
This property name is a little unclear. Is intended to be this a file path? Fully-qualified or not? Consider using `FileInfo` if so.

Reply:
Renamed to CacheFilePath since this contains the full path to the file.

I will change this to FileInfo if needed.

Reply:
`FileInfo` is better because it's more specific and safer. `string` won't enforce anything about the content, and also brings the `null` versus `string.Empty` confusion.